code/datums/components/moved_relay.dm

```diff
@@ -0,0 +1,66 @@
+/**
+ * This component will ensure that move calls from parent locs
+ * will be relayed to the parent atom.
+ * This means that something inside of another object can
+ * still know when that other object moves.
+ */
+/datum/component/moved_relay
+	//List of ordered parents
+	//Index 1: parent
+	//Index 2: Parent's parent
+	//etc.
+	var/list/atom/ordered_parents = list()
+
+/datum/component/moved_relay/Initialize(...)
+	var/atom/A = parent
+	//Start tracking from the parent
+	//We will relay the parents move to itself for convenience
+	//if the super parent gets deleted, everything dies
+	RegisterSignal(A, COMSIG_PARENT_QDELETING, .proc/component_deleted)
+	RegisterSignal(A, COMSIG_MOVABLE_MOVED, .proc/parent_moved)
+	ordered_parents += A
+	//Recursively register parents
+	if(!isturf(A.loc))
+		register_parent(A.loc)
+
+/datum/component/moved_relay/Destroy(force, silent)
+	while(length(ordered_parents))
+		unregister_parent(ordered_parents[1])
+	. = ..()
+
+/datum/component/moved_relay/proc/register_parent(atom/A)
+	RegisterSignal(A, COMSIG_PARENT_QDELETING, .proc/parent_deleted)
```

Comment:
Components are deleted when their parents are. Why is this needed over just putting it in Destroy()?

Reply:
register_parent is recursive, it registers the parent and the parent's parent too. The one that could be changed would be the original signal registering on initialise.